WordPress/Classes/Utility/Migration/ContentMigrationCoordinator.swift

```diff
@@ -0,0 +1,105 @@
+/// Encapsulates logic related to content migration from WordPress to Jetpack.
+///
+class ContentMigrationCoordinator {
+
+    static let shared: ContentMigrationCoordinator = .init()
```

Comment:
In the next PR, the `ContentMigrationCoordinator` will attempt to sync local post drafts first. Also, adding a dependency to `PostService`. This will happen asynchronously and without a shared instance (or someone that holds a reference to this instance). The `completion` block of the methods will be garbage collected.

I can put the shared implementation in the later PR, but I figured it's better to do it now, so I don't have to change the other files twice (Jetpack overlay sheets, AppDelegate).